pkg/apis/v1alpha1/ipaddressallocation_types.go

```diff
@@ -0,0 +1,62 @@
+/* Copyright © 2023 VMware, Inc. All Rights Reserved.
+   SPDX-License-Identifier: Apache-2.0 */
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	VisibilityExternal = "External"
```

Comment:
Should we define three constants for Subnet AccessMode and use the same here? How modes are defined in NSX API?

Reply:
https://github.com/vmware-tanzu/nsx-operator/blob/main/pkg/apis/v1alpha1/subnet_types.go#L20-L20
Here is Private or Public, we need External, Private or Project, how about define a new constant?

Reply:
First, in NSX API spec, do we have a common type definition for IP/IPBlock visibility and Subnet modes? Does NSX API use "Public" for Subnets? @lxiaopei

Reply:
In NSX API, IPBlock and ip_address_block_visibility are using `External`, access mode of Subnet is using `Public`.

Reply:
Ok, in this case, let us add a new type for IPAddressVisibility. I guess NSX API may keep changing and let us watch the changes.

Reply:
ok